fix(server-status): enable module in debian if `server_status_require…

javierbertoli opened the pull request:
…` is set

<!--
Please fill in this PR template to make it easier to review and merge.

It has been designed so that a lot of it can be completed *after* submitting it,
e.g. filling in the checklists.

Notes:
1. Please keep the PR as small as is practicable; the larger a PR gets, the harder it becomes to review and the more time it requires to get it merged.
2. Similarly, please avoid PRs that cover more than one type; it should be a _bug fix_ *OR* a _new feature_ *OR* a _refactor_, etc.
3. Please direct questions to the [`#formulas` channel on Slack](https://saltstackcommunity.slack.com/messages/C7LG8SV54/), which is bridged to `#saltstack-formulas` on Freenode.
4. Feel free to suggest improvements to this template by reporting an issue or submitting a PR.  The source of this template is:
  - https://github.com/saltstack-formulas/.github/blob/master/.github/pull_request_template.md
-->

### PR progress checklist (to be filled in by reviewers)
<!-- Please leave this checklist for reviewers to tick as they work through the PR. -->

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

### What type of PR is this?
<!-- Please tick each box that is relevant (after creating the PR). -->

#### Primary type
<!-- There really should be only *one* of these types ticked for each PR. -->

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type
<!-- Most PRs should include all of the following types as well. -->

- [ ] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?
<!-- If so, change the following to a `Yes` and explain what the breaking changes are. -->
<!-- If there are multiple breaking changes, list them all. -->

No.

### Related issues and/or pull requests
<!-- Please link any related issues/PRs here, especially any issues that are closed by this PR. -->



### Describe the changes you're proposing
<!-- A clear and concise description of what you have implemented. -->
<!-- Consider explaining each commit if they cover different aspects of the proposed changes. -->

`server-status` management was added to the formula a while ago (a3c0022d), but was not for Debian. This PR adds this.

### Pillar / config required to test the proposed changes
<!-- Provide links to the SLS files and/or relevant configs (be sure to remove sensitive info). -->

### Debug log showing how the proposed changes work
<!-- Include a debug log showing how these changes work, e.g. using `salt-minion -l debug`. -->
<!-- Alternatively, linking to Kitchen debug logs is useful, e.g. via. Travis CI. -->
<!-- Most useful is providing a passing InSpec test, which can be used to verify any proposed changes. -->



### Documentation checklist
<!-- Please tick each box that is relevant (after creating the PR). -->

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist
<!-- Please tick each box that is relevant (after creating the PR). -->

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### Additional context
<!-- Add any other context about the proposed changes here. -->

baby-gnu commented on apache/server_status.sls:
```diff
@@ -18,3 +18,25 @@ include:
       - module: apache-reload
       - service: apache
+
+{% if grains['os_family']=="Debian" %}
+a2endisconf server-status:
+  cmd.run:
+{%   if apache.get('server_status_require') is defined %}
+    - name: a2enconf server-status
+    - unless: test -L /etc/apache2/conf-enabled/server-status.conf
+{%   else %}
+    - name: a2disconf server-status
+    - onlyif: test -L /etc/apache2/conf-enabled/server-status.conf
+{%   endif %}
```
As a matter of style, is it possible to declare some variable to [avoid putting jinja conditionals within the states as much as possible](https://docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#light-conditionals-and-looping)?

For example:

``` jinja
{%- if grains['os_family']=="Debian" %}
{%-   if apache.get('server_status_require') is defined %}
{%-     set cmd = 'a2enconf server-status' %}
{%-     set cond = {'unless': 'test -L /etc/apache2/conf-enabled/server-status.conf'} %}
{%-   else %}
{%-     set cmd = 'a2disconf server-status' %}
{%-     set cond = {'onlyif': 'test -L /etc/apache2/conf-enabled/server-status.conf'} %}
{%-   endif %}
a2endisconf server-status:
  cmd.run:
    - name: {{ cmd }}
    - {{ cond }}
```

javierbertoli replied:
> I'm fine with the changes.
> 
> I'm wondering what the `if apache.get('server_status_require') is defined` does when there is no `apache:server_status_require`, does the `is defined` handle the `KeyError`?

It does. Anyway, will refactor the code and add some tests also.

baby-gnu replied:
> It does. Anyway, will refactor the code and add some tests also.

What do you think about this instead:

```jinja
{%-   if apache.get('server_status_require', False) %}
```

Otherwise, if the user set `apache:server_status_require` to `false` it will apply the configuration regardless.

javierbertoli replied:
Actually, checking that logic, I removed that piece of code because it's wrong: as is, it would **disable** `server-status` when no _`server_status_require` entries_ are set. But the module, by default, is usually enabled with _localhost_ allowed and therefore these entries are optional.

myii commented on test/integration/default/controls/packages_spec.rb:
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# Overide by OS
+package_name = 'bash'
+package_name = 'cronie' if (os[:name] == 'centos') && os[:release].start_with?('6')
+
```
```suggestion
```

javierbertoli replied:
ooops, :smile:

myii commented on test/integration/default/controls/packages_spec.rb:
```diff
@@ -0,0 +1,23 @@
+      'apache2'
+    when 'redhat', 'fedora'
+      'httpd'
+    when 'arch'
```
As discussed in the #formulas channel yesterday, `arch` is under the `linux` family:

```suggestion
    # `linux` here is sufficient for `arch`
    when 'linux'
```

myii commented on test/integration/modules/controls/server_status_spec.rb:
```diff
@@ -0,0 +1,35 @@
+      '/etc/httpd/conf.d'
+    when 'suse'
+      '/etc/apache2/conf.d'
+    when 'arch'
```
Same as above.

myii commented:
@javierbertoli Just noticed: all of the tests cover the families `debian`, `redhat`, `fedora` & `suse`.  Only some of them cover `arch` (i.e. `linux`).  So the other ones won't work on `arch`.  We should have this consistent, right?

javierbertoli commented:
@myii you're absolutely right. Fixed that and pushed, only a single `mod_security` test is missing now `Arch` but will try to fix that in another PR: I notice that when I wrote the `mod_security` module I added an extra yaml file which, now, makes no sense. I'd like to refactor the formula a little there.

myii commented:
Merged, thanks for this @javierbertoli; appreciate the review @baby-gnu.

Just a note that matrix [needs to be updated](https://github.com/myii/ssf-formula/pull/240/files#diff-9331ce86567ed9f40bda29728e82c97bR500) to use the new `module` suite added here.

saltstack-formulas-travis commented:
:tada: This PR is included in version 0.41.1 :tada:

The release is available on [GitHub release](https://github.com/saltstack-formulas/apache-formula/releases/tag/v0.41.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
